File: src/2465.number-of-distinct-averages.2561/solution.ts
function distinctAverages (nums: number[]): number {
    const set = new Set();
    nums.sort((a, b) => a - b);
    let left = 0;
    let right = nums.length - 1;
    while (left < right) {
        const avg = (nums[left++] + nums[right--]) / 2;
        set.add(avg);
    }
    return set.size;
}

File: src/2465.number-of-distinct-averages.2561/solution.java
class Solution {
    public int distinctAverages(int[] nums) {
        Set<Double> set = new HashSet<>();
        Arrays.sort(nums);
        int l = 0;
        int r = nums.length-1;
        while (l<r){
            set.add( (nums[l++]+nums[r--])/2.0 );
        }
        return set.size();
    }
}